apps/dutchclock/app.js

```diff
@@ -0,0 +1,267 @@
+
+const SCREEN_WIDTH = g.getWidth();
+const SCREEN_HEIGHT = g.getHeight();
+
+const TOP_SPACING = 5;
+const WIDGETS_HEIGHT = 20;
+const DATETIME_SPACING_HEIGHT = 5;
+const TIME_HEIGHT = 10;
+const DATE_HEIGHT = 10;
+const BOTTOM_SPACING = 5;
+
+const TEXT_WIDTH = SCREEN_WIDTH - 2;
+
+const MINS_IN_HOUR = 60;
+const MINS_IN_DAY = 24 * MINS_IN_HOUR;
+
+const VARIANT_EXACT = 'exact';
+const VARIANT_APPROXIMATE = 'approximate';
+const VARIANT_HYBRID = 'hybrid';
+
+const DEFAULTS_FILE = "dutchclock.default.json"; 
+const SETTINGS_FILE = "dutchclock.json";
+
+// Load settings
+const settings = Object.assign(
+    storage.readJSON(DEFAULTS_FILE, true) || {},
+    storage.readJSON(SETTINGS_FILE, true) || {}
+);
+
+const maxFontSize = SCREEN_HEIGHT 
+  - TOP_SPACING
+  - (settings.showWidgets ? WIDGETS_HEIGHT : 0)
+  - (settings.showDate || settings.showTime ? DATETIME_SPACING_HEIGHT : 0)
+  - (settings.showDate ? DATE_HEIGHT : 0)
+  - (settings.showTime ? TIME_HEIGHT : 0);
+
+const X = SCREEN_WIDTH / 2;
+const Y = SCREEN_HEIGHT / 2
+  + TOP_SPACING / 2
+  + (settings.showWidgets ? WIDGETS_HEIGHT / 2 : 0)
+  - (settings.showDate || settings.showTime ? DATETIME_SPACING_HEIGHT / 2 : 0)
+  - (settings.showDate ? DATE_HEIGHT / 2 : 0)
+  - (settings.showTime ? TIME_HEIGHT / 2 : 0);
+  
+let date, mins;
+
+function initialize() {
+  // draw immediately at first
+  tick();
+
+  // now check every second
+  let secondInterval = setInterval(tick, 1000);
+  // Stop updates when LCD is off, restart when on
+
+  Bangle.on('lcdPower',on=>{
+    if (secondInterval) clearInterval(secondInterval);
+    secondInterval = undefined;
+    if (on) {
+      secondInterval = setInterval(tick, 1000);
+      draw(); // draw immediately
+    }
+  });
+}
+
+function tick() {
+  date = new Date();
+  const m = (date.getHours() * MINS_IN_HOUR + date.getMinutes()) % MINS_IN_DAY;
+
+  if (m !== mins) {
+    mins = m;
+    draw();
+  }
+
+  if (!settings.showWidgets) {
+    widget_utils.hide();
+  }
+}
+
+function draw() {
+  // work out how to display the current time
+  const timeLines = getTimeLines(mins);
+  const bottomLines = getBottomLines();
+
+  // Reset the state of the graphics library
+  g.clear(true);
+
+  // draw the current time (4x size 7 segment)
+  setFont(timeLines);
+
+  g.setFontAlign(0,0); // align center top
+  g.drawString(timeLines.join("\n"), X, Y, false);
+
+  if (bottomLines.length) {  
+    // draw the time and/or date, in a normal font
+    g.setFont("6x8");
+    g.setFontAlign(0,1); // align center bottom
+    // pad the date - this clears the background if the date were to change length
+    g.drawString(bottomLines.join('\n'), SCREEN_WIDTH/2, SCREEN_HEIGHT - BOTTOM_SPACING, false);
+  }
+
+  /* Show launcher when middle button pressed
+  This should be done *before* Bangle.loadWidgets so that
+  widgets know if they're being loaded into a clock app or not */
+  Bangle.setUI("clock");
+  // Load widgets
+  Bangle.loadWidgets();
+
```

Comment:
Can this be moved into the initialize function? Or is there a reason for it being here?